Pocket/pocket-ios
# [WIP] IN-718: Filter Items by Tag

## Summary
Feature Branch that includes the following work:
- Add View to Filter My List / Archive by Existing Tags and `not tagged`
- Add Tag Labels to Item Cells 
- Add Tagged Topic Chip when user is filtering by an item

Note: Editing and Renaming Tags will be handled in another PR

## References 
IN-718, IN-719, IN-786, IN-792

## Implementation Details
[WIP]

## Test Steps
- [ ] Given that the user is on My List or Archive, 
when the saved item has a tag, 
then they should be able to view the tag label on the list cell.

- [ ] Given that the user is on My List or Archive, 
when the saved item has a tag, 
then they should be able to view at max two tag labels on the list cell. 

- [ ] Given that a saved item has a tag, 
when the user taps on the label, 
then they should see the filter view for that tag.

- [ ] GIVEN I see the tagged bottom sheet
AND I haven’t selected a tag
WHEN I dismiss the bottom sheet without selecting a tag
THEN I am sent to “All”

- [ ] WHEN I click the tagged button
THEN I see a bottom sheet with a list of all tags

- [ ] WHEN I click a tag in the bottom sheet
THEN the bottom sheet disappears
AND I see the list filtered by the selected tag

- [ ] “not tagged” tag is always on the tag list at the top
It shows up right under the filter carousel when selected (like the other tags)
It is a permanent tag

- [ ] Given the user filters their list by a tag, 
and the Tagged topic chip is highlighted, 
then we should see a selected tag chip (as shown in screenshot).

- [ ] Given the user sees the selected tag chip,
and the user taps on the close button in the chip, 
then it removes the tag filter and goes back to “All”. 

## PR Checklist:

## Screenshots
https://user-images.githubusercontent.com/6743397/188678335-22da4fd0-558a-4616-b7c6-389c47023712.mp4
